docs/topics/workers.md

```diff
@@ -1,18 +1,117 @@
 exposes a host of Node.js libraries to `brigade.js` files, as well as the
 `brigadier` Brigade library.
 
-Sometimes, though, it is desirable to add additional libraries -- perhaps even
-custom libraries -- to your workers. This guide walks through the process of
-adding custom libraries.
+Sometimes, though, it is desirable to include additional libraries -- perhaps even
+custom libraries -- to your workers. 
+There are two ways of adding custom dependencies to a Brigade worker:
+
+- by adding a `brigade.json` file in your repository (similar to a `package.json` file) 
+that contains the dependencies and that are specific on every Brigade project
+
+- by creating a custom Docker image for the worker that already contains the dependencies 
+and which will be used for all Brigade projects
+
+This guide explains both processes and describes the cases where each method is most suitable.
 
 **Note:** This area of Brigade is still evolving. If you have ideas for improving
 it, feel free to [file an issue](https://github.com/Azure/brigade/issues) explaining
 your idea.
 
+## Add custom dependencies using a `brigade.json` file
+
+If you need different dependencies for every Brigade project, this can be easily achieved 
+using a `brigade.json` file placed side-by-side the `brigade.js` file. This file contains 
+the dependency name and version, and has the following structure:
+
+```
+{
+    "dependencies": {
+        "is-thirteen": "2.0.0"
+    }
+}
+```
+Before starting to execute the `brigade.js` script, the worker will install the  
+dependencies using `yarn`, adding them to the `node_modules` folder.
+
+Then, in the `brigade.js` file, the new dependency can be used just like any 
+other NodeJS dependency:
+
+```
+const { events } = require("brigadier")
+const is = require("is-thirteen");
+
+events.on("exec", function (e, p) {
+    console.log("is 13 thirteen? " + is(13).thirteen());
+})
+```
+
+Now if we run a build for this project, we see the `is-thirteen` dependency added, 
+as well as the console log resulted from using it:
+
+```
+$ brig run brigade-86959b08a89af5b6b83f0ace6d9030f1fdca7ed8ea0a296e27d72e
+Event created. Waiting for worker pod named "brigade-worker-01cexrvrs08shcev26961cwd6n".
+Started build 01cexrvrs08shcev26961cwd6n as "brigade-worker-01cexrvrs08shcev26961cwd6n"
+installing is-thirteen@2.0.0
+prestart: src/brigade.js written
+[brigade] brigade-worker version: 0.14.0
+[brigade:k8s] Creating PVC named brigade-worker-01cexrvrs08shcev26961cwd6n
+is 13 thirteen? true
+[brigade:app] after: default event handler fired
+[brigade:app] beforeExit(2): destroying storage
+[brigade:k8s] Destroying PVC named brigade-worker-01cexrvrs08shcev26961cwd6n
+```
+
+Notes:
+
+- the dependencies should point the exact version (and not use the tilde ~ and caret ^ to indicate semver compatible versions)
+
+- note the already existing dependencies and dev dependencies for a Brigade worker:
```

Comment:
these dependencies will change over time as we update brigade-worker's package.json so this file may need to be constantly updated. Would it be better to link to https://github.com/Azure/brigade/blob/master/brigade-worker/package.json#L21-L38 instead?

Reply:
Yeah, that's a great idea!
I will probably add some dependencies as examples in the doc though, but will link to the actual file from master.

Reply:
that sounds like a good compromise. 👍